src/game/Main.cpp

```diff
@@ -44,7 +44,15 @@ LRESULT CALLBACK WndProcMain(HWND hWnd, UINT message, WPARAM wParam, LPARAM lPar
 				if(hwndCtl == CN3UIEdit::s_hWndEdit)
 				{
 					pEdit->UpdateTextFromEditCtrl();
-					pEdit->UpdateCaretPosFromEditCtrl();
+					if (GetAsyncKeyState(VK_TAB) || GetAsyncKeyState(VK_LBUTTON)) {) {
+						pEdit->UpdateCaretPosFromEditCtrl(true);
+					}
+					else {
+						pEdit->UpdateCaretPosFromEditCtrl(false);
+					}
+
+
+					
 					CGameProcedure::SetGameCursor(CGameProcedure::s_hCursorNormal);
```

Comment:
There is syntax error here and it fails to compile: `{)`. We can also write it this way to improve readability:
```suggestion
					pEdit->UpdateTextFromEditCtrl();
					bool bNeedsFocus = GetAsyncKeyState(VK_TAB) || GetAsyncKeyState(VK_LBUTTON);
					pEdit->UpdateCaretPosFromEditCtrl(bNeedsFocus);
					CGameProcedure::SetGameCursor(CGameProcedure::s_hCursorNormal);
```